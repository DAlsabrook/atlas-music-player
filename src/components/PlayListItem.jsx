export function PlayListItem (props) {
    return (
        <div className={`w-full flex justify-between items-center mb-2 rounded-md ${props.isPlaying ? 'bg-(--secondary) dark:bg-(--bg-color) text-white dark:text-(--secondary)' : 'bg-transparent'}`}>
            <div className="flex flex-col">
                <p className='font-bold'>{props.songName}</p>
                <p className='font-bold opacity-70'>{props.artist}</p>
            </div>
            <p className="font-bold opacity-50">{props.songTime}</p>
        </div>
    )
}
