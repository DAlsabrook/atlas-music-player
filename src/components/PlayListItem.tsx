type PlayListItemProps = {
  songName: string;
  songTime: string;
  artist: string;
  isPlaying: boolean;
}



export function PlayListItem({ songName, songTime, artist, isPlaying }: PlayListItemProps) {

    return (
        <div className={`w-full flex justify-between items-center mb-2 rounded-md ${isPlaying ? 'bg-(--secondary) dark:bg-(--bg-color) text-white dark:text-(--secondary)' : 'bg-transparent'}`}>
            <div className="flex flex-col">
                <p className='font-bold'>{songName}</p>
                <p className='font-bold opacity-70'>{artist}</p>
            </div>
            <p className="font-bold opacity-50">{songTime}</p>
        </div>
    )
}
