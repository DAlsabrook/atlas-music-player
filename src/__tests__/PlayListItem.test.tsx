import { afterEach, expect, test } from "vitest";
import { cleanup, render } from "@testing-library/react";
import { PlayListItem } from "@/components/PlayListItem.tsx";

// Cleans DOM and any change that may have happened in a test
afterEach(() => {
    cleanup();
});

async function fetchPlaylist() {
    const response = await fetch('/api/playlist');
    return response.json();
}

test('Playing is false', async () => {
    const playlist = await fetchPlaylist();
    const container = render(<PlayListItem songName={playlist.title} songTime={playlist.duration} artist={playlist.artist} isPlaying={false}/>)
    expect(container).toMatchSnapshot();
})

test('Playing is true', async () => {
    const playlist = await fetchPlaylist();
    const container = render(<PlayListItem songName={playlist.title} songTime={playlist.duration} artist={playlist.artist} isPlaying={true}/>)
    expect(container).toMatchSnapshot();
})

test('Duration is too long', async () => {
    const playlist = await fetchPlaylist();
    const container = render(<PlayListItem songName={playlist.title} songTime={12345678} artist={playlist.artist} isPlaying={true}/>)
    expect(container).toMatchSnapshot();
})

test('Renders with minimal props', () => {
    const container = render(<PlayListItem songName="Test Song" songTime={120} artist="Test Artist" isPlaying={false}/>)
    expect(container).toMatchSnapshot();
})
// @ts-ignore
test('Handles missing props', () => {
    const container = render(<PlayListItem songName="Test Song" songTime={120} artist="Test Artist" />)
    expect(container).toMatchSnapshot();
})
// @ts-ignore
test('Handles typo in props', () => {
    const container = render(<PlayListItem songName="Test Song" songTime={120} artist="Test Artist" isplayin={true} />)
    expect(container).toMatchSnapshot();
})

test('Renders with different durations', async () => {
    const playlist = await fetchPlaylist();
    const container = render(<PlayListItem songName={playlist.title} songTime={300} artist={playlist.artist} isPlaying={false}/>)
    expect(container).toMatchSnapshot();
})
