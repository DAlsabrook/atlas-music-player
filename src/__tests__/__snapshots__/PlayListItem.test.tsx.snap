// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Duration is too long 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex justify-between items-center mb-2 rounded-md bg-(--secondary) dark:bg-(--bg-color) text-white dark:text-(--secondary)"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="font-bold"
          />
          <p
            class="font-bold opacity-70"
          />
        </div>
        <p
          class="font-bold opacity-50"
        >
          1:2345678
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex justify-between items-center mb-2 rounded-md bg-(--secondary) dark:bg-(--bg-color) text-white dark:text-(--secondary)"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="font-bold"
        />
        <p
          class="font-bold opacity-70"
        />
      </div>
      <p
        class="font-bold opacity-50"
      >
        1:2345678
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Handles missing props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="font-bold"
          >
            Test Song
          </p>
          <p
            class="font-bold opacity-70"
          >
            Test Artist
          </p>
        </div>
        <p
          class="font-bold opacity-50"
        >
          1:20
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="font-bold"
        >
          Test Song
        </p>
        <p
          class="font-bold opacity-70"
        >
          Test Artist
        </p>
      </div>
      <p
        class="font-bold opacity-50"
      >
        1:20
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Handles typo in props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="font-bold"
          >
            Test Song
          </p>
          <p
            class="font-bold opacity-70"
          >
            Test Artist
          </p>
        </div>
        <p
          class="font-bold opacity-50"
        >
          1:20
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="font-bold"
        >
          Test Song
        </p>
        <p
          class="font-bold opacity-70"
        >
          Test Artist
        </p>
      </div>
      <p
        class="font-bold opacity-50"
      >
        1:20
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Playing is false 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="font-bold"
          />
          <p
            class="font-bold opacity-70"
          />
        </div>
        <p
          class="font-bold opacity-50"
        >
          0:
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="font-bold"
        />
        <p
          class="font-bold opacity-70"
        />
      </div>
      <p
        class="font-bold opacity-50"
      >
        0:
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Playing is true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex justify-between items-center mb-2 rounded-md bg-(--secondary) dark:bg-(--bg-color) text-white dark:text-(--secondary)"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="font-bold"
          />
          <p
            class="font-bold opacity-70"
          />
        </div>
        <p
          class="font-bold opacity-50"
        >
          0:
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex justify-between items-center mb-2 rounded-md bg-(--secondary) dark:bg-(--bg-color) text-white dark:text-(--secondary)"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="font-bold"
        />
        <p
          class="font-bold opacity-70"
        />
      </div>
      <p
        class="font-bold opacity-50"
      >
        0:
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders with different durations 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="font-bold"
          />
          <p
            class="font-bold opacity-70"
          />
        </div>
        <p
          class="font-bold opacity-50"
        >
          3:00
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="font-bold"
        />
        <p
          class="font-bold opacity-70"
        />
      </div>
      <p
        class="font-bold opacity-50"
      >
        3:00
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders with minimal props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="font-bold"
          >
            Test Song
          </p>
          <p
            class="font-bold opacity-70"
          >
            Test Artist
          </p>
        </div>
        <p
          class="font-bold opacity-50"
        >
          1:20
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex justify-between items-center mb-2 rounded-md bg-transparent"
    >
      <div
        class="flex flex-col"
      >
        <p
          class="font-bold"
        >
          Test Song
        </p>
        <p
          class="font-bold opacity-70"
        >
          Test Artist
        </p>
      </div>
      <p
        class="font-bold opacity-50"
      >
        1:20
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
